Week1/homework/app.js

```diff
@@ -2,10 +2,130 @@
 {
   const bookTitles = [
     // Replace with your own book titles
-    'harry_potter_chamber_secrets'
+    'harry_potter_chamber_secrets',
+    'desert_solitaire',
+    'geek_love',
+    'gilead',
+    'disgrace',
+    'infinite_jest',
+    'lolita',
+    'maus',
+    'poems',
+    'things_fall_apart'
   ];
+  console.log(bookTitles);
 
+  //2.3 
+
+  /*  // hold this function  based on request 2.5 
+
+  function createAndAppend(titles) {
+    const ul = document.createElement("ul"),
+      root = document.getElementById("root");
+    root.appendChild(ul);
+
+    for (let i = 0; i < titles.length; i++) {
+      const li = document.createElement("li");
+      ul.appendChild(li);
+      li.setAttribute('id', titles[i]);
+    }
+  }
+
+  createAndAppend(bookTitles);
+
+  */
+
+  // 2.4  
+  const booksDetails = {
+    'harry_potter_chamber_secrets': { title: 'harry potter chamber secrets', language: "English", author: "J. K. Rowling" },
+    'desert_solitaire': { title: 'desert solitaire', language: "English", author: "Edward Abbey" },
+    'geek_love': { title: 'geek love', language: "English", author: "Katherine Dunn" },
+    'gilead': { title: 'gilead', language: "English", author: "Marilynne Robinson" },
+    'disgrace': { title: 'disgrace', language: "English", author: "J. M. Coetzee" },
+    'infinite_jest': { title: 'infinite jest', language: "English", author: "David Foster Wallace" },
+    'lolita': { title: 'lolita', language: "English", author: "Vladimir Nabokov" },
+    'maus': { title: 'maus', language: "English", author: "Art Spiegelman" },
+    'poems': { title: 'poems', language: "English", author: "Elizabeth Bishop" },
+    'things_fall_apart': { title: 'things fall apart', language: "English", author: "Chinua Achebe" }
+  };
+
+  // just check
+  console.log(booksDetails);
+  for (const key in booksDetails) {
+    console.log(key);
+    const x = booksDetails[key];
+    console.log(x);
+
+    for (const prop in x) {
+      console.log(prop + " : " + x[prop]);
+    }
+  }
```

Comment:
A PR is an explicit request to others to review your code. This code is for debugging. It should not be in a PR. You don't want to burden reviewers with having to find out if code is relevant for the functioning of an application. If you want to keep this code for your own purposes, that's fine. Then for the homework you could create a new branch in which you remove all this debug code and use that for the PR.